tests/unit/draw/ElkLayout.spec.js

```diff
@@ -0,0 +1,494 @@
+import { expect, describe, jest } from '@jest/globals';
+
+import ElkLayout from 'src/draw/ElkLayout';
+import Component from 'src/models/Component';
+import ComponentDefinition from 'src/models/ComponentDefinition';
+import ComponentAttribute from 'src/models/ComponentAttribute';
+import ComponentDrawOption from '../../../src/models/ComponentDrawOption';
+
+const exampleComponents = [new Component({
+  id: 'workflow1',
+  name: 'verticalWorkflow',
+  definition: new ComponentDefinition({ parentTypes: ['network', 'workflow'] }),
+  drawOption: new ComponentDrawOption({
+    x: 292,
+    y: 12,
+    width: 308,
+    height: 406,
+  }),
+  attributes: [],
+}), new Component({
+  id: 'wfstep1',
+  name: 'step1',
+  definition: new ComponentDefinition({ parentTypes: ['workflow'] }),
+  drawOption: new ComponentDrawOption({
+    x: 30,
+    y: 30,
+    width: 230,
+    height: 50,
+  }),
+  attributes: [new ComponentAttribute({
+    name: 'workflow',
+    value: 'workflow1',
+    definition: {
+      type: 'Reference',
+      name: 'workflow',
+    },
+  })],
+}), new Component({
+  id: 'wfstep2',
+  name: 'step2',
+  definition: new ComponentDefinition({ parentTypes: ['workflow'] }),
+  drawOption: new ComponentDrawOption({
+    x: 30,
+    y: 110,
+    width: 230,
+    height: 50,
+  }),
+  attributes: [new ComponentAttribute({
+    name: 'workflow',
+    value: 'workflow1',
+    definition: {
+      type: 'Reference',
+      name: 'workflow',
+    },
+  })],
+})];
+
+const exampleLinks = [{
+  source: 'wfstep1',
+  target: 'wfstep2',
+}];
+
+describe('Test Class: ElkLayout', () => {
+  describe('Test static field elk', () => {
+    it('should have imported elkjs', () => {
+      expect(typeof ElkLayout.elk.layout).toEqual('function');
+    });
+  });
+
+  describe('Test method: getNodes', () => {
+    const layoutManager = new ElkLayout(undefined);
+
+    it('should throw exception on undefined argument', () => {
```

Comment:
Useless test, because components can't be undefined. Always equals to an array.

Reply:
removed